packages/libs/src/sdk/services/Solana/programs/PaymentRouterClient/PaymentRouterClient.ts

```diff
@@ -0,0 +1,251 @@
+
+  private readonly mints: Prettify<Partial<Record<Mint, PublicKey>>>
+
+  private existingTokenAccounts: Prettify<Partial<Record<Mint, Account>>>
+
+  constructor(config: PaymentRouterClientConfig) {
+    const configWithDefaults = mergeConfigWithDefaults(
+      config,
+      getDefaultPaymentRouterClientConfig(productionConfig)
+    )
+    super(configWithDefaults, config.solanaWalletAdapter)
+    this.programId = configWithDefaults.programId
+    const [pda, bump] = PublicKey.findProgramAddressSync(
+      [new TextEncoder().encode('payment_router')],
+      this.programId
+    )
+    this.programAccount = pda
+    this.programAccountBumpSeed = bump
+    this.mints = configWithDefaults.mints
+    this.existingTokenAccounts = {}
+  }
+
+  public async createTransferInstruction(
+    params: CreateTransferInstructionRequest
+  ) {
+    const args = await parseParams(
+      'crateTransferInstruction',
+      CreateTransferInstructionSchema
+    )(params)
+    const mint = this.mints[args.mint]
+    if (!mint) {
+      throw Error('Mint not configured')
+    }
+    const programTokenAccount = await this.getOrCreateProgramTokenAccount({
+      mint: args.mint
+    })
+    const owner = this.wallet.publicKey
+    if (!owner) {
+      throw Error('Connected wallet not found.')
+    }
+    const sourceTokenAccount = getAssociatedTokenAddressSync(mint, owner, false)
+    const amount = mintFixedDecimalMap[args.mint](args.amount)
+    return createTransferCheckedInstruction(
+      sourceTokenAccount,
+      mint,
+      programTokenAccount.address,
+      owner,
+      amount.value,
+      amount.decimalPlaces
+    )
+  }
+
+  public async createRouteInstruction(params: CreateRouteInstructionRequest) {
+    const args = await parseParams(
+      'createRouteInstruction',
+      CreateRouteInstructionSchema
+    )(params)
+    const programTokenAccount = await this.getOrCreateProgramTokenAccount({
+      mint: args.mint
+    })
+    const recipients: PublicKey[] = []
+    const amounts: bigint[] = []
+    for (const [key, value] of Object.entries(args.splits)) {
+      recipients.push(new PublicKey(key))
+      amounts.push(value)
+    }
+    const totalAmount = mintFixedDecimalMap[args.mint](args.total).value
+    return PaymentRouterProgram.createRouteInstruction({
+      sender: programTokenAccount.address,
+      senderOwner: this.programAccount,
+      paymentRouterPdaBump: this.programAccountBumpSeed,
+      recipients,
+      amounts,
+      totalAmount,
+      programId: this.programId
+    })
+  }
+
+  public async createMemoInstruction(params: CreateMemoInstructionRequest) {
+    const {
+      contentType,
+      contentId,
+      blockNumber,
+      buyerUserId,
+      accessType,
+      signer
+    } = await parseParams(
+      'createMemoInstructionSchema',
+      CreateMemoInstructionSchema
+    )(params)
+    const memoString = `${contentType}:${contentId}:${blockNumber}:${buyerUserId}:${accessType}`
```

Comment:
nit: This function is generically named but producing a very specific output.
* Can we add docs for this?
* Should it be more explicitly named? (`createPurchaseMemoInstruction`?)